### PR TITLE
build(tsc): Use angular2-template-compiler in place of tsc

### PR DESCRIPTION
**Pull request by alexeagle**
This lets us down-level Decorators with tsickle and produce .metadata.json
files for users to reference when offline-compiling their app.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "angular": "^1.5.0",
     "angular-animate": "^1.5.0",
     "angular-mocks": "^1.5.0",
+    "angular2-template-compiler": "^0.1.5",
```
Comment (tbosch) [inline package.json]: Alex, shouldn't the template compiler that works with the new world use the `@angular` namespace as well for the package name?

File: npm-shrinkwrap.json [hunk 2]
```diff
@@ -8549,6 +8559,11 @@
         }
       }
     },
+    "ts-metadata-collector": {
+      "version": "0.1.1",
+      "from": "ts-metadata-collector@>=0.1.0 <0.2.0",
+      "resolved": "file:dist/tools/metadata"
```
Comment (robwormald) [inline npm-shrinkwrap.json]: probably not right here

File: npm-shrinkwrap.json [hunk 3]
```diff
@@ -88,6 +88,16 @@
...
     },
+    "angular2": {
+      "version": "2.0.0-beta.17",
+      "from": "angular/angular#2.0.0-build.cacdead.js",
+      "resolved": "git://github.com/angular/angular.git#08cbd5481d04255864f5ff8e33d0756d4ec4c440"
+    },
+    "angular2-template-compiler": {
+      "version": "0.1.5",
+      "from": "angular2-template-compiler@0.1.5",
+      "resolved": "file:dist/js/cjs/compiler_cli"
```
Comment (robwormald) [inline npm-shrinkwrap.json]: also breaks npm install

**Comment by tbosch** [comment]
@alexeagle We need a change in the `reflection_capabilities.ts` to accept the slightly different metadata format:

1. `ReflectionCapabilities.parameters(type)`: Has to look for `typeOfFunc.ctorParameters` and `typeOfFunc.decorators` using the format used in tsickle and merge these two. Especially it needs to map decorator classes + their args to metadata instances (very similar to what the `StaticReflector` does).
2. `ReflectionCapabilities.annotations(type)`: Has to look for `typeOfFunc.decorators` using the format used in tsickle. Especially it needs to map decorator classes + their args to metadata instances (very similar to what the `StaticReflector` does).
3. `ReflectionCapabilities.propMetadata(type)`: Has to look for `typeOfFunc.propDecorators` using the format in tsickle. Especially it needs to map decorator classes + their args to metadata instances (very similar to what the `StaticReflector` does).

File: package.json (hunk 1 above)
Comment (alexeagle) [inline package.json]: Yes, we should do that at some point during the launch, and deprecate this package. Will sync with Igor on this tomorrow in SLC

File: npm-shrinkwrap.json (hunk 2 above)
Comment (alexeagle) [inline npm-shrinkwrap.json]: oh, interesting artifact from npm link I guess?

**Comment by alexeagle** [comment]
will be in #8380

**Pull request closed without merging**